### PR TITLE
Fix Bug 1437875: Assorted docker updates

### PR DESCRIPTION
**Pull request by phlax**
This provides most of the fixes/updates from #867 without using dockerhub for the builds - which we can look at separately

Specifically, this PR does the following:

- dont copy app directory into image (this was creating huge images with my unclean env) and isnt really necessary if you are always running out of a local app dir
- dont run docker build from root directory to keep the build context nice and small (requires getting requirements from http)
- runs webpack/collectstatic automatically in dockerrun (fixes Bug 1437875)

### CODE REVIEW

File: docker/dev/config/webapp.env [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+DJANGO_DEV=True
+DJANGO_DEBUG=True
+DATABASE_URL=postgres://pontoon:asdf@postgresql/pontoon
+MANUAL_SYNC=True
+SESSION_COOKIE_SECURE=False
+SITE_URL=http://localhost:8000
+FXA_CLIENT_ID=2651b9211a44b7b2
+FXA_SECRET_KEY=a3cafccbafe39db54f2723f8a6f804c337e362950f197b5b33050d784129d570
+FXA_OAUTH_ENDPOINT=https://oauth-stable.dev.lcip.org/v1
+FXA_PROFILE_ENDPOINT=https://stable.dev.lcip.org/profile/v1
```
[the remark this answers is not in the document]
Comment (phlax) [reply docker/dev/config/webapp.env]: ah, k - ill remove

File: docker/dev/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# Setup apps's env
+COPY ./set_up_env.sh /app/
+RUN sudo -u app /app/set_up_env.sh
+
+# Install Pontoon Python requirements
+# gets from github to keep the build context nice and small
+ADD https://raw.githubusercontent.com/mozilla/pontoon/master/requirements.txt /app/requirements.txt
+ADD https://raw.githubusercontent.com/mozilla/pontoon/master/requirements-dev.txt /app/requirements-dev.txt
+ADD https://raw.githubusercontent.com/mozilla/pontoon/master/requirements-test.txt /app/requirements-test.txt
```
[the remark this answers is not in the document]
Comment (phlax) [reply docker/dev/Dockerfile]: hmm - im wondering if you should have to rebuild the image to do that tho

[the remark this answers is not in the document]
Comment (adngdb) [reply docker/dev/Dockerfile]: That's what I would assume. And that's how I work right now, every time I make a change to requirements, I run `make dockerbuild`. I would expect the same for JS dependencies btw, but that's out of scope here.

File: docker/dev/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+RUN pip install -U 'pip>=8'
+RUN pip install --no-cache-dir --require-hashes -r requirements-dev.txt
+RUN pip install --no-cache-dir --require-hashes -r requirements-test.txt
+
+# COPY . /app/
```
[the remark this answers is not in the document]
Comment (adngdb) [reply docker/dev/Dockerfile]: Ah, but of course, it has a volume! Stupid me forgot how he built this stuff... :P

[the remark this answers is not in the document]
Comment (phlax) [reply docker/dev/Dockerfile]: the problem is the context - docker is a client/daemon even for building so the filesystem is sent to the daemon at the point of context - in this case it was actually using it - but at the cost of making large images

**Comment by adngdb** [comment]
Hey @phlax, I'm closing this in favor of #930. I've been looking at what others at Mozilla are doing, and it's quite different from the direction you were going to here. I'd rather stay consistent with what Mozilla projects are doing. See https://github.com/mozilla-services/tecken https://github.com/mozilla-services/antenna and https://github.com/mozilla-services/socorro for my main references.

**Pull request closed without merging**